src/ApplicationInsights.Kubernetes/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -1,29 +1,98 @@
+            this TelemetryConfiguration telemetryConfiguration,
+            TimeSpan? timeout = null,
+            IKubernetesServiceCollectionBuilder kubernetesServiceCollectionBuilder = null)
+        {
+            if (telemetryConfiguration == null)
+            {
+                return;
+            }
+
+            IServiceCollection standaloneServiceCollection = new ServiceCollection();
+            standaloneServiceCollection = EnableKubernetesImpl(standaloneServiceCollection, timeout, kubernetesServiceCollectionBuilder);
+
+            IServiceProvider serviceProvider = standaloneServiceCollection.BuildServiceProvider();
+
+            ITelemetryInitializer k8sTelemetryInitializer = serviceProvider.GetServices<ITelemetryInitializer>()
+                .First(ti => ti.GetType() == typeof(KubernetesTelemetryInitializer));
+
+            telemetryConfiguration.TelemetryInitializers.Add(k8sTelemetryInitializer);
+        }
+
+        /// <summary>
+        /// Enable applicaiton insights for kubernetes.
+        /// </summary>
+        private static IServiceCollection EnableKubernetesImpl(IServiceCollection serviceCollection,
             TimeSpan? timeout = null,
             IKubernetesServiceCollectionBuilder kubernetesServiceCollectionBuilder = null)
         {
-            // Dispatch this on a differnet thread to avoid blocking the main thread.
-            // Mainly used with K8s Readness Probe enabled, where communicating with Server will temperory be blocked.
-            // TODO: Instead of query the server on the start, we should depend on watch services to provide dynamic realtime data.
-            Task.Run(() =>
+            // 2 minutes by default maximum to wait for spinning up the container.
+            timeout = timeout ?? TimeSpan.FromMinutes(2);
+
+            // According to the code, adding logging will not overwrite existing logging classes
+            // https://github.com/aspnet/Logging/blob/c821494678a30c323174bea8056f43b93a3ca6f4/src/Microsoft.Extensions.Logging/LoggingServiceCollectionExtensions.cs
+            // Becuase it uses 'TryAdd()' extenion method on service collection.
+            serviceCollection.AddLogging();
+
+            IServiceProvider serviceProvider = serviceCollection.BuildServiceProvider();
+            ILogger logger = serviceProvider.GetService<ILogger<IServiceCollection>>();
+            logger.LogInformation(Invariant($"ApplicationInsights.Kubernetes.Version:{SDKVersionUtils.Instance.CurrentSDKVersion}"));
+            try
+            {
+                serviceCollection = BuildK8sServiceCollection(serviceCollection, timeout.Value, kubernetesServiceCollectionBuilder);
+            }
+            catch (Exception ex)
             {
-                KubernetesModule.EnableKubernetes(services, timeout, kubernetesServiceCollectionBuilder);
-            });
+                logger.LogError("Failed to fetch ApplicaitonInsights.Kubernetes' version info. Details" + ex.ToString());
```

Comment:
(minor) add space after "Details" in the error message

Are you sure the problem will always be fetching AppInsights.K8s version info, and not something else?